userspace/libsinsp/test/sinsp_metrics.ut.cpp

```diff
@@ -0,0 +1,382 @@
+
+	// Another round of fake metric tests since we do not fetch real scap metrics, for example.
+	std::vector<metrics_v2> fake_metrics_snapshot;
+	fake_metrics_snapshot.emplace_back(libs_metrics_collector->new_metric("sys_enter.run_cnt",
+											METRICS_V2_LIBBPF_STATS, 
+											METRIC_VALUE_TYPE_U64,
+											METRIC_VALUE_UNIT_COUNT, 
+											METRIC_VALUE_MONOTONIC, 
+											76435525241UL));
+
+	fake_metrics_snapshot.emplace_back(libs_metrics_collector->new_metric("sys_enter.run_time_ns",
+											METRICS_V2_LIBBPF_STATS, 
+											METRIC_VALUE_TYPE_U64,
+											METRIC_VALUE_UNIT_TIME_NS_COUNT, 
+											METRIC_VALUE_MONOTONIC, 
+											16269369826392UL));
+
+	fake_metrics_snapshot.emplace_back(libs_metrics_collector->new_metric("sys_enter.avg_time_ns",
+											METRICS_V2_LIBBPF_STATS, 
+											METRIC_VALUE_TYPE_U64,
+											METRIC_VALUE_UNIT_TIME_NS, 
+											METRIC_VALUE_NON_MONOTONIC_CURRENT, 
+											203UL));
+
+	fake_metrics_snapshot.emplace_back(libs_metrics_collector->new_metric("n_drops",
+											METRICS_V2_KERNEL_COUNTERS, 
+											METRIC_VALUE_TYPE_U64,
+											METRIC_VALUE_UNIT_COUNT, 
+											METRIC_VALUE_MONOTONIC, 
+											674200UL));
+
+	fake_metrics_snapshot.emplace_back(libs_metrics_collector->new_metric("n_drops_buffer_total",
+											METRICS_V2_KERNEL_COUNTERS, 
+											METRIC_VALUE_TYPE_U64,
+											METRIC_VALUE_UNIT_COUNT, 
+											METRIC_VALUE_MONOTONIC, 
+											5000UL));
+
+	// Simulate some derived metrics; critical for example for Falco consumer use cases
+	fake_metrics_snapshot.emplace_back(libs_metrics_collector->new_metric("duration_sec",
+											METRICS_V2_MISC,
+											METRIC_VALUE_TYPE_U64,
+											METRIC_VALUE_UNIT_TIME_S_COUNT, 
+											METRIC_VALUE_MONOTONIC, 
+											144UL));
+
+	fake_metrics_snapshot.emplace_back(libs_metrics_collector->new_metric("evt_rate_sec",
+											METRICS_V2_MISC,
+											METRIC_VALUE_TYPE_D,
+											METRIC_VALUE_UNIT_TIME_S, 
+											METRIC_VALUE_NON_MONOTONIC_CURRENT, 
+											126065.4));
+
+	// Timestamps while they always go up should still be regarded as gauge from a Prometheus perspective
+	// https://www.robustperception.io/are-increasing-timestamps-counters-or-gauges/
+	fake_metrics_snapshot.emplace_back(libs_metrics_collector->new_metric("host_boot_ts",
+											METRICS_V2_MISC, 
+											METRIC_VALUE_TYPE_U64,
+											METRIC_VALUE_UNIT_TIME_TIMESTAMP_NS, 
+											METRIC_VALUE_NON_MONOTONIC_CURRENT, 
+											1708753667000000000UL));
+
+	for (auto& metric: fake_metrics_snapshot)
+	{
+		prometheus_metrics_converter->convert_metric_to_unit_convention(metric);
+		prometheus_text = prometheus_metrics_converter->convert_metric_to_text(metric, "testns", "falco");
+		printf("%s", prometheus_text.c_str());
+		if (strncmp(metric.name, "sys_enter.run_cnt", 18) == 0)
```

Comment:
`strncmp` has the length parameter - as opposed to plain `strcmp` - to protect from memory errors; reading outside the buffer, in this case. Using the length of the string literal, that is by construction zero terminated, is not going to help here. You should use the length of `metric.name`. Here and in all subsequent ones. Otherwise just use `strcmp` and avoid the magic number.